torchaudio/transforms.py

```diff
@@ -907,3 +909,313 @@ def forward(self, waveform: Tensor) -> Tensor:
         cmn_waveform = F.sliding_window_cmn(
             waveform, self.cmn_window, self.min_cmn_window, self.center, self.norm_vars)
         return cmn_waveform
+
+
+class Vad(torch.nn.Module):
+    r"""Voice Activity Detector. Similar to SoX implementation.
+    Attempts to trim silence and quiet background sounds from the ends of recordings of speech.
+    The algorithm currently uses a simple cepstral power measurement to detect voice,
+    so may be fooled by other things, especially music.
+
+    The effect can trim only from the front of the audio,
+    so in order to trim from the back, the reverse effect must also be used.
+
+    Args:
+        trigger_level (float, optional): The measurement level used to trigger activity detection.
+            This may need to be cahnged depending on the noise level, signal level,
+            and other characteristics of the input audio. (Default: 7.0)
+        trigger_time (float, optional): The time constant (in seconds)
+            used to help ignore short bursts of sound. (Default: 0.25)
+        search_time (float, optional): The amount of audio (in seconds)
+            to search for quieter/shorter bursts of audio to include prior
+            to the detected trigger point. (Default: 1.0)
+        allowed_gap (float, optional): The allowed gap (in seconds) between
+            quiteter/shorter bursts of audio to include prior
+            to the detected trigger point. (Default: 0.25)
+        pre_trigger_time (float, optional): The amount of audio (in seconds) to preserve
+            before the trigger point and any found quieter/shorter bursts. (Default: 0.0)
+
+    References:
+        http://sox.sourceforge.net/sox.html
+        https://pysox.readthedocs.io/en/latest/api.html#sox.transform.Transformer.vad
+    """
+
+    @dataclass
+    class Channel:
```

Comment:
If the information for each channel is held in a class, it will be more difficult to vectorize, no?

Reply:
That's right! I am gonna experiment, refactor and report back.

Reply:
> That's right. I can technically pull the state inside the computation, declaring classes inside the function - let's see how it's gonna affect the readability and vectorization.

The state you refer to is the channel class, right? That would be one more reason to avoid implementing it :)

Reply:
Thanks for your feedback @vincentqb and @mthrok. Got rid of `dataclass` and state class as a whole.